R/lifecycle_admiral.R

```diff
@@ -0,0 +1,46 @@
+#' Deprecation with Soft Message
+#'
+#' Wrapper around `lifecycle::deprecate_soft()`.
+#'
+#' @param env See documentation for `lifecycle::deprecate_soft()`.
+#'
+#' @param user_env See documentation for `lifecycle::deprecate_soft()`.
+#'
+#' @param ... See documentation for `lifecycle::deprecate_soft()` for additional argument use.
+#'
+#'
+#'
+#' @return `NULL`, invisibly.
+#'
+#' @examples
+#' # A Phase 1 deprecated function with custom bulleted list:
+#' deprecate_inform(
+#'   when = "1.0.0",
+#'   what = "foo()",
+#'   details = c(
+#'     #     'x = "This message will turn into a warning with release of x.y.z",
+#'     i = "See admiral's deprecation guidance:
+#' https://pharmaverse.github.io/admiraldev/dev/articles/programming_strategy.html#deprecation"
+#'   )
+#' )
+#'
+#' @keywords messages
+#' @family messages
+#'
+#'
+#' @export
+deprecate_inform <- function(
+    env = caller_env(),
+    user_env = caller_env(2),
+    ...) {
+  tryCatch(
+    lifecycle::deprecate_soft(
+      env = caller_env(),
+      user_env = caller_env(2),
```

Comment:
These should be passed from the argument values, instead of hard-coded, right?

We could change the definition to this:

```r
deprecate_inform <- function( when,
                              what,
                              with = NULL,
                              details = NULL,
                              id = NULL,
                              env = caller_env(),
                              user_env = caller_env(2)) {
  tryCatch(
    lifecycle::deprecate_soft(
      when = when,
      what = what,
      with = with,
      details = details,
      id = id,
      env = env,
      user_env = user_env
    ),
    warning = \(w) {
      message(conditionMessage(w))
      tryInvokeRestart("muffleWarning")
    }
  )
}
```


I had only used `...` in our Slack chat, because I was being lazy (if that is the origin of the `...`)

Reply:
ah yes - i thought we were doing `...` to allow for passing of additional arguments for something else later on.  will update soon :)